packages/aws-serverless/scripts/buildLambdaLayer.ts

```diff
@@ -61,3 +64,113 @@ function fsForceMkdirSync(path: string): void {
   fs.rmSync(path, { recursive: true, force: true });
   fs.mkdirSync(path);
 }
+
+async function pruneNodeModules(): Promise<void> {
+  const entrypoints = [
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/esm/index.js',
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/cjs/index.js',
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/cjs/awslambda-auto.js',
+    './build/aws/dist-serverless/nodejs/node_modules/@sentry/aws-serverless/build/npm/esm/awslambda-auto.js',
+  ];
+
+  const { fileList } = await nodeFileTrace(entrypoints);
+
+  const allFiles = getAllFiles('./build/aws/dist-serverless/nodejs/node_modules');
+
+  const filesToDelete = allFiles.filter(file => !fileList.has(file));
+  console.log(`Removing ${filesToDelete.length} unused files from node_modules.`);
+
+  for (const file of filesToDelete) {
+    try {
+      fs.unlinkSync(file);
+    } catch {
+      console.error(`Error deleting ${file}`);
+    }
+  }
+
+  console.log('Cleaning up empty directories.');
+
+  removeEmptyDirs('./build/aws/dist-serverless/nodejs/node_modules');
```

Comment:
is it important/necessary that we clean up empty dirs? Just wondering...
Alternate idea, not sure if this makes sense: instead of deleting the files that are not in nft, could we instead just copy the files that _are_ traced by nft into a final place? 🤔